control/freqplot.py

```diff
@@ -1039,14 +1025,238 @@ def gangof4_plot(P, C, omega=None, **kwargs):
 
     plt.tight_layout()
 
+#
+# Singular values plot
+#
+
+
+def singular_values_plot(syslist, omega=None,
+                         plot=True, omega_limits=None, omega_num=None,
+                         *args, **kwargs):
+    """Singular value plot for a system
+
+    Plots a Singular Value plot for the system over a (optional) frequency range.
+
+    Parameters
+    ----------
+    syslist : linsys
+        List of linear systems (single system is OK).
+    omega : array_like
+        List of frequencies in rad/sec to be used for frequency response.
+    plot : bool
+        If True (default), generate the singular values plot.
+    omega_limits : array_like of two values
+        Limits of the frequency vector to generate.
+        If Hz=True the limits are in Hz otherwise in rad/s.
+    omega_num : int
+        Number of samples to plot.
+        Default value (1000) set by config.defaults['freqplot.number_of_samples'].
+    dB : bool
+        If True, plot result in dB.
+        Default value (False) set by config.defaults['freqplot.dB'].
+    Hz : bool
+        If True, plot frequency in Hz (omega must be provided in rad/sec).
+        Default value (False) set by config.defaults['freqplot.Hz']
+
+    Returns
+    -------
+    sigma : ndarray (or list of ndarray if len(syslist) > 1))
+        singular values
+    omega : ndarray (or list of ndarray if len(syslist) > 1))
+        frequency in rad/sec
+
+    Other Parameters
+    ----------------
+    grid : bool
+        If True, plot grid lines on gain and phase plots.  Default is set by
+        `config.defaults['freqplot.grid']`.
+
+    Examples
+    --------
+    >>> import numpy
```

Comment:
```suggestion
    >>> import numpy as np
```

Please, either import under the name as it is used later, or leave the import out (together with `singular_values_plot` and change the `tf` call to `TransferFunction`), because the namespace of the module itself is available during a `pytest --doctest-modules` call.

Reply:
done, I kept the numpy as np import and used TransferFunction instead of tf